code/ExcludeChildren.php

```diff
@@ -22,30 +22,33 @@ public function getExcludedClasses(){
 		return $this->hiddenChildren;
 	}
 	
-	public function getFilteredChildren($children){
+	public function getFilteredChildren($children) {
 		// Optionally force exclusion beyond CMS (eg. exclude from $Children as well)
-		if( $this->owner->config()->get("force_exclusion_beyond_cms") 
-				&& get_class(Controller::curr()) != "CMSPagesController" // default CMS controller
-				&& get_class(Controller::curr()) != "CMSPageEditController"){ // default CMS edit controller
-			return $children->exclude('ClassName', $this->getExcludedClasses());
-		}
-		// Else; exclude children from CMS only
-		$action = Controller::curr()->getAction();
-		if (in_array($action, array("treeview", "listview", "getsubtree"))) {
+		$controller = Controller::curr();
+		$action = $controller->getAction();
+		if ($this->owner->config()->get("force_exclusion_beyond_cms")
+			|| ($controller instanceof LeftAndMain) && in_array($action, array("treeview", "listview", "getsubtree"))
+		) {
+			//if the page class has a getExcludedChildren function, use it to supply the list of children
+			if ($this->owner->hasMethod('getExcludedChildren')) {
```

Comment:
getExcludedChildren is misleading as we're actually returning the one that we're not excluding here